### ISSUE TITLE
System upgrade process MUST be improved

### ISSUE THREAD

**Issue by jaimestuardo** [magento/magento2#3084]
Hello....

This is a request for URGENT improvements about system upgrade. I spent almost the whole day to upgrade from version 2.0.0 to 2.0.1. After several errors and rollbacks it was unable to upgrade.

The first exception was the following, which does not have any sense since Magento is trying to access user root directory:

[ErrorException] 
chdir(): open_basedir restriction in effect. File(/home/depasa) is not within the allowed path(s): (/home/depasa/public_html:/tmp:/home/depasa/tmpsite:/opt/php5-6/lib/php:/opt/ferozo/etc/suspen:/opt/ferozo/suspended.page)

After solving error above by adding the root directory to allowed directories in php.ini, an error occured while upgrading.

The error was due to disabled functions I had for my systems for security reasons. For instance, escapeshellarg function was disabled.

When that error occured, I could not even rollback. Fortunately, I have another installation in my server, so I copied whole vendor folder and I could finally roll back.

I think the above condition must be trapped in the Readiness check.

When I solved those errors, another error was shown:

[ErrorException] 
Source /home/depasa/public_html/vendor/magento/magento2-base/dev/tests/integration/tmp does not exist

Then I have manually created tmp folder.

When the last error occured, bin folder was empty, so I recovered it first and same as before, I needed to restore all content of vendor folder too.

Then, I tried to upgrade once more, however, it was impossible to upgrade this time. When I went to setup option, this apache error was shown in logs:

zend_mm_heap corrupted
[Fri Jan 22 13:59:15.904689 2016] [fastcgi:error] [pid 5752](104)Connection reset by peer: [client 190.215.3.76:9578] FastCGI: comm with server "/opt/apache/cgi-sys/depasa/php-56" aborted: read failed, referer: http://www.depasa.cl/setup/
[Fri Jan 22 13:59:15.904802 2016] [fastcgi:error] [pid 5752] [client 190.215.3.76:9578] FastCGI: incomplete headers (0 bytes) received from server "/opt/apache/cgi-sys/depasa/php-56", referer: http://www.depasa.cl/setup/

I needed to restart webserver in order to be able to proceed.

Then, for my suprise, the same error about missing tmp directory was shown.

In summary, upgrade process is really a disaster that needs URGENT correction.

Regards
Jaime

**Comment by tanya-soroka** [comment]
@jaimestuardo , We have created ticket MAGETWO-48259 to investigate.

**Comment by buskamuza** [comment]
@jaimestuardo , we're investigating the issue.
Do you still have full log from the cases when upgrade failed?
It would be really helpful to understand on which steps this happens. Under steps I mean the upgrade process itself (it can be on a stage of downloading the code or on applying SQL upgrades or something else), so the log may help us with this question.

Thanks in advance.

**Comment by jaimestuardo** [comment]
Where can I find a log of this process?

**Comment by buskamuza** [comment]
It should be in `var/update_status.log`.

**Comment by jaimestuardo** [comment]
There you are. This is the whole file that shows all tries of upgrading.

[update_status.txt](https://github.com/magento/magento2/files/101468/update_status.txt)

**Comment by buskamuza** [comment]
Thanks a lot!

**Comment by buskamuza** [comment]
@jaimestuardo , few more questions:
1. How did you install 2.0.0? Is it from an archive?
2. Do you have `dev/tests/integration/tmp` folder in Magento root?

**Comment by jaimestuardo** [comment]
1. From an archive and then with the web installation wizard
2. I have created manually the tmp folder where the log says: /home/depasa/public_html/vendor/magento/magento2-base/dev/tests/integration/.

When upgrading, tmp folder was gone.

![magento-dir-integration](https://cloud.githubusercontent.com/assets/9195659/12524162/f93d6b12-c139-11e5-9406-a6e6a6cebfac.png)

**Comment by Jedwards-Eric** [comment]
Yeah my upgrade also broke and stalled.  Waste of many hours and I still haven't fixed it.  Does Magento have a public QA team to test these updates prior to release?  If yes I'm will to join to just to avoid this type of situation.

**Comment by sshrewz** [comment]
This issue has been fixed and merged to develop.  Feel free to reopen if you encounter further problems.
